rbx_binary/src/text_deserializer.rs

```diff
@@ -286,6 +290,51 @@ impl DecodedValues {
 
                 Some(DecodedValues::Axes(values))
             }
+            Type::CFrame => {
+                let mut rotations = vec![Matrix3::identity(); prop_count];
+
+                for i in 0..prop_count {
+                    let id = reader.read_u8().unwrap();
+                    if id == 0 {
+                        rotations[i] = Matrix3::new(
+                            Vector3::new(
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                            ),
+                            Vector3::new(
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                            ),
+                            Vector3::new(
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                                reader.read_f32::<LittleEndian>().unwrap(),
+                            ),
+                        );
+                    } else {
+                        rotations[i] = special_case_to_rotation(id).unwrap();
+                    }
+                }
+
+                let mut x = vec![0.0; prop_count];
+                let mut y = vec![0.0; prop_count];
+                let mut z = vec![0.0; prop_count];
+
+                reader.read_interleaved_f32_array(&mut x).unwrap();
+                reader.read_interleaved_f32_array(&mut y).unwrap();
+                reader.read_interleaved_f32_array(&mut z).unwrap();
+
+                let mut values =
+                    vec![CFrame::new(Vector3::new(0.0, 0.0, 0.0), Matrix3::identity()); prop_count];
+
+                for i in 0..prop_count {
+                    values[i] = CFrame::new(Vector3::new(x[i], y[i], z[i]), rotations[i])
+                }
```

Comment:
Same sort of deal with iterator zipping here as above. This code is quite a bit shorter and more readable than the iterator zipping code, I wonder if we can discern whether there's a perf loss from the assembly...

Reply:
The zipping code turns out to not be that much less readable, so I think I'm okay with switching to it; the performance benefit of not indexing 5 vectors is probably worth it.